orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/model/support/TriggerDeserializer.kt

```diff
@@ -0,0 +1,127 @@
+          get("dryRun")?.booleanValue() == true,
+          get("strategy")?.booleanValue() == true,
+          get("master").textValue(),
+          get("job").textValue(),
+          get("buildNumber").intValue(),
+          get("propertyFile")?.textValue()
+        ).apply {
+          buildInfo = get("buildInfo")?.parseValue(parser)
+          properties = get("properties")?.mapValue(parser) ?: mutableMapOf()
+        }
+        looksLikePipeline() -> PipelineTrigger(
+          get("type").textValue(),
+          get("correlationId")?.textValue(),
+          get("user")?.textValue() ?: "[anonymous]",
+          get("parameters")?.mapValue(parser) ?: mutableMapOf(),
+          get("artifacts")?.listValue(parser) ?: mutableListOf(),
+          get("notifications")?.listValue(parser) ?: mutableListOf(),
+          get("rebake")?.booleanValue() == true,
+          get("dryRun")?.booleanValue() == true,
+          get("strategy")?.booleanValue() == true,
+          get("parentExecution").parseValue<Execution>(parser),
+          get("parentPipelineStageId")?.textValue()
+        )
+        looksLikeGit() -> GitTrigger(
+          get("type").textValue(),
+          get("correlationId")?.textValue(),
+          get("user")?.textValue() ?: "[anonymous]",
+          get("parameters")?.mapValue(parser) ?: mutableMapOf(),
+          get("artifacts")?.listValue(parser) ?: mutableListOf(),
+          get("notifications")?.listValue(parser) ?: mutableListOf(),
+          get("rebake")?.booleanValue() == true,
+          get("dryRun")?.booleanValue() == true,
+          get("strategy")?.booleanValue() == true,
+          get("source").textValue(),
+          get("project").textValue(),
+          get("branch").textValue(),
+          get("slug").textValue()
+        )
+        else -> DefaultTrigger(
+          get("type").textValue(),
+          get("correlationId")?.textValue(),
+          get("user")?.textValue() ?: "[anonymous]",
+          get("parameters")?.mapValue(parser) ?: mutableMapOf(),
+          get("artifacts")?.listValue(parser) ?: mutableListOf(),
+          get("notifications")?.listValue(parser) ?: mutableListOf(),
+          get("rebake")?.booleanValue() == true,
+          get("dryRun")?.booleanValue() == true,
+          get("strategy")?.booleanValue() == true
+        )
+      }.apply {
+        other = mapValue(parser)
+        resolvedExpectedArtifacts = get("resolvedExpectedArtifacts")?.listValue(parser) ?: mutableListOf()
+      }
+    }
+
+  private fun JsonNode.looksLikeDocker() =
+    hasNonNull("account") && hasNonNull("repository") && hasNonNull("tag")
+
+  private fun JsonNode.looksLikeGit() =
+    hasNonNull("source") && hasNonNull("project") && hasNonNull("branch") && hasNonNull("slug")
+
+  private fun JsonNode.looksLikeJenkins() =
+    hasNonNull("master") && hasNonNull("job") && hasNonNull("buildNumber")
+
+  private fun JsonNode.looksLikePipeline() =
+    hasNonNull("parentExecution")
```

Comment:
Above methods determine type by the fields present in the JSON, ignoring the `type` field.